### PR TITLE
use httpclient to power HTTPUtilities class

### PR DESCRIPTION
**Pull request by egh**
Second try at this. Changes the exceptions raised slightly but should be basically the same.

### CODE REVIEW

File: spec/http_utilities_spec.rb [hunk 1]
```diff
@@ -117,7 +117,7 @@ module RichCitationsProcessor
 
         it "should raise an exception" do
           stub_request(http_method, 'http://www.example.com/path').to_return(status:404).times(1)
-          expect { action('http://www.example.com/path', 'application/custom') }.to raise_exception(Net::HTTPServerException)
+          expect(action('http://www.example.com/path', 'application/custom')).to be_nil
```
Comment (marcrohloff) [inline spec/http_utilities_spec.rb]: The test no longer describes what is being tested

File: spec/http_utilities_spec.rb [hunk 2]
```diff
@@ -131,7 +131,7 @@ module RichCitationsProcessor
           expect(HTTPUtilities).to receive(:sleep).with(15)
           expect(HTTPUtilities).to_not receive(:sleep)
 
-          expect { action('http://www.example.com/path') }.to raise_exception(Net::HTTPFatalError)
+          expect(action('http://www.example.com/path')).to be_nil
```
Comment (marcrohloff) [inline spec/http_utilities_spec.rb]: Does returning nil make sense? How does the code know if there was an error or not?

**Comment by egh** [comment]
Good points, Marc. I've addressed the issues.

**Comment by egh** [comment]
Thanks!

**Pull request merged**